cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/CombinedHBaseMetricsTable.java

```diff
@@ -158,25 +136,41 @@ public long incrementAndGet(byte[] row, byte[] column, long delta) {
   @Override
   public void delete(byte[] row, byte[][] columns) {
     v3HBaseTable.delete(row, columns);
-    if (tableExists.apply(resolution)) {
+    try {
       v2HBaseTable.delete(row, columns);
```

Comment:
this will always throw NPE after v2Table was deleted. You need to check whether it is null.